src/main/java/world/bentobox/bentobox/managers/island/IslandGrid.java

```diff
@@ -12,59 +11,40 @@
  *
  */
 class IslandGrid {
-    private final TreeMap<Integer, TreeMap<Integer, Island>> grid = new TreeMap<>();
-    private final BentoBox plugin = BentoBox.getInstance();
+    private final TreeMap<Integer, TreeMap<Integer, String>> grid = new TreeMap<>();
+    private final IslandCache im;
+
+    /**
+     * @param im IslandsManager
+     */
+    public IslandGrid(IslandCache im) {
+        super();
+        this.im = im;
+    }
 
     /**
      * Adds island to grid
      * @param island - island to add
      * @return true if successfully added, false if island already exists, or there is an overlap
      */
     public boolean addToGrid(Island island) {
+        // Check if we know about this island already
         if (grid.containsKey(island.getMinX())) {
-            TreeMap<Integer, Island> zEntry = grid.get(island.getMinX());
+            TreeMap<Integer, String> zEntry = grid.get(island.getMinX());
             if (zEntry.containsKey(island.getMinZ())) {
-                // There is an overlap or duplicate
```

Comment:
A lot of old defensive code has been removed. If it's required, it can be added again, but I don't think it's needed.